algorithms/linfa-preprocessing/benches/whitening_bench.rs

```diff
@@ -1,38 +1,47 @@
+use criterion::{black_box, criterion_group, criterion_main, BenchmarkId, Criterion};
+use linfa::benchmarks::config;
 use linfa::traits::Fit;
 use linfa::traits::Transformer;
+use linfa_datasets::generate::make_dataset;
 use linfa_preprocessing::whitening::Whitener;
-use ndarray::Array2;
-use ndarray_rand::{
-    rand::distributions::Uniform, rand::rngs::SmallRng, rand::SeedableRng, RandomExt,
-};
+use statrs::distribution::{DiscreteUniform, Laplace};
 
-fn iai_pca_bench() {
-    let mut rng = SmallRng::seed_from_u64(21);
-    for nfeatures in (10..100).step_by(10) {
-        fit_transform_whitener(Whitener::pca(), &mut rng, 10000, nfeatures);
-    }
-}
-
-fn iai_zca_bench() {
-    let mut rng = SmallRng::seed_from_u64(21);
-    for nfeatures in (10..100).step_by(10) {
-        fit_transform_whitener(Whitener::zca(), &mut rng, 10000, nfeatures);
-    }
-}
+fn bench(c: &mut Criterion) {
+    let mut benchmark = c.benchmark_group("whitening");
+    config::set_default_benchmark_configs(&mut benchmark);
+    let size = 10000;
+    let feat_distr = Laplace::new(0.5, 5.).unwrap();
+    let target_distr = DiscreteUniform::new(0, 5).unwrap();
 
-fn iai_cholesky_bench() {
-    let mut rng = SmallRng::seed_from_u64(21);
-    for nfeatures in (10..100).step_by(10) {
-        fit_transform_whitener(Whitener::cholesky(), &mut rng, 10000, nfeatures);
+    for (whitener, fn_name) in [
+        (Whitener::cholesky(), "cholesky"),
+        (Whitener::zca(), "zca"),
+        (Whitener::pca(), "pca"),
+    ] {
+        for nfeatures in (10..100).step_by(10) {
+            let dataset = make_dataset(size, nfeatures, 1, feat_distr, target_distr);
+            benchmark.bench_function(
+                BenchmarkId::new(fn_name, format!("{}x{}", nfeatures, size)),
+                |bencher| {
+                    bencher.iter(|| {
+                        whitener
+                            .fit(black_box(&dataset.clone()))
+                            .unwrap()
+                            .transform(black_box(dataset.clone()));
```

Comment:
I think you can put `dataset.view()` here to prevent unnecessary clone

Reply:
Done